### PR TITLE
Fix explosive bow knockback being way too high if close to the entity

### PR DESCRIPTION
**Pull request by Sfiguz7**
## Description

As the code is now, the Explosive Bow seems to work fine as long as there is enough distance from the impact point and nearby entities, especially if they are on the same level. When shooting an entity from close by though, especially if directly above, you get shot in the air even 40-60 blocks up which is definitely not supposed to happen.

## Proposed changes

- Cleaned up the extra redundant calculations
- Modified knockback calculation to achieve reasonable pushes
## Related Issues (if applicable)

N/A
## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by Sfiguz7** [comment]
This has been way more of a pain than it should've. Special thanks to @svr333 for testing, @Sefiraat for commenting random stuff in my test builds and @WalshyDev for being extra picky on the values

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/ExplosiveBow.java [hunk 1]
```diff
@@ -56,21 +56,20 @@ public BowShootHandler onShoot() {
             for (Entity nearby : entites) {
                 LivingEntity entity = (LivingEntity) nearby;
 
-                Vector distanceVector = entity.getLocation().toVector().subtract(target.getLocation().toVector());
-                distanceVector.setY(distanceVector.getY() + 0.6);
-
-                Vector velocity = entity.getVelocity();
+                Vector distanceVector = entity.getLocation().toVector().subtract(target.getLocation().toVector())
+                    .add(new Vector(0, 0.75, 0));
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/ExplosiveBow.java:60]: Format consistentlyy
```suggestion
                Vector distanceVector = entity.getLocation().toVector()
                    .subtract(target.getLocation().toVector())
                    .add(new Vector(0, 0.75, 0));
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/ExplosiveBow.java [hunk 2]
```diff
@@ -56,21 +56,20 @@ public BowShootHandler onShoot() {
...
 
                 double distanceSquared = distanceVector.lengthSquared();
-                double damage = calculateDamage(distanceSquared, e.getDamage());
+                double damage = e.getDamage() * (1 - (distanceSquared / (2 * range.getValue() * range.getValue())));
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/ExplosiveBow.java:63]: We should definitely max this to some value.

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/ExplosiveBow.java (hunk 2 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/ExplosiveBow.java:64]: It is, the second term is between 0.5 and 1 so the entire thing is between e.getDamage() and e.getDamage()/2

**Review by svr333 — APPROVED** [review]

**Review by WalshyDev — APPROVED** [review]
LGTM

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/ExplosiveBow.java (hunk 2 above)
Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/ExplosiveBow.java:64]: neeeeeeeeeeeeeeeeeeerrrrrrrrrrrrrrrrrrrrrrrrrrrrddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd

**Pull request merged**